### PR TITLE
feat(wdl-cli): infer call prefix

### PR DESCRIPTION
**Pull request by a-frantz**
[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.1.0/

### CODE REVIEW

File: wdl-cli/src/inputs.rs [hunk 1]
```diff
@@ -235,12 +240,15 @@ impl Inputs {
     }
 
     /// Attempts to coalesce a set of inputs into an [`Inputs`].
-    pub fn coalesce<T, V>(iter: T) -> Result<Self>
+    pub fn coalesce<T, V>(iter: T, name: Option<String>) -> Result<Self>
     where
```
Comment (acfoltzer) [inline wdl-cli/src/inputs.rs:244]: The docstring needs a description of `name`. I also notice that none of the tests within this repo use a `name` argument other than `None`. I realize that it is exercised with https://github.com/stjude-rust-labs/sprocket/pull/147 but could you please add some tests so we can have some local assurance and examples of the behavior?

**Review by acfoltzer — CHANGES_REQUESTED** [review]

File: wdl-engine/tests/inputs/missing-prefix/error.txt
[the remark this answers is not in the document]
Comment (a-frantz) [reply wdl-engine/tests/inputs/missing-prefix/error.txt:1]: The file needs to be there and be empty I believe

[the remark this answers is not in the document]
Comment (peterhuene) [reply wdl-engine/tests/inputs/missing-prefix/error.txt:1]: Ah right, this test keeps an empty errors file if there isn't an error.

File: wdl-cli/src/inputs.rs (hunk 1 above)
Comment (a-frantz) [reply wdl-cli/src/inputs.rs]: `coalesce` is essentially just `Inputs::new` and the `name` member is only accessed in `into_engine_inputs()` which isn't actually tested anywhere in `wdl-cli` at this point. I could add new tests that would use the new `libtest-mimic`? Don't think I'd finish that before I sign off for the day, but we can hold this off till tomorrow.

File: wdl-engine/src/inputs.rs [hunk 2]
```diff
@@ -829,26 +829,30 @@ impl Inputs {
...
                          workflow or task name",
                     )
                 }
             },
-            // If the object is empty, treat it as a workflow evaluation without any inputs
             None => {
-                return Ok(None);
+                unreachable!()
             }
         };
```
[the remark this answers is not in the document]
Comment (a-frantz) [reply wdl-engine/src/inputs.rs]: reverted the block 👍

**Review by acfoltzer — APPROVED** [review]
